### ISSUE TITLE
PubSub: messages with group topics are sent many times per node

### ISSUE THREAD

**Issue by yufei-cai** [akka/akka#27421]
Setup:
- 100 actors in actor system A subscribe to 1 topic with groups distinct from one another.
- 1 message is published in actor system B on the topic with `sendOneMessageToEachGroup=true`.

Expected behavior:
1 message is sent from B to A, where it gets broadcasted to the 100 subscribers.

Actual behavior:
100 messages are sent from B to A.

The actual behavior is a logical conclusion of the documentation (assuming "topic"==="topic identifier":  "the message is sent over the wire only once per node (that has a matching topic)" and "the group id is used it is part of the topic identifier", desto the optimization does not apply for message published to multiple groups). Is the actual behavior reasonable? Does the documentation leaves the right impression? I believed the "expected behavior" for years...

**Comment by patriknw** [comment]
It’s supposed to send the once over the wire so your understanding of expected behavior is correct. Not sure if the ”group” aspect changes any of that. Have to look into the details, but on vacation now.

**Comment by patriknw** [comment]
It looks like when using groups that optimization is not used. I don't know how difficult it would be to improve this, and it's not something we can work on right now. I hope you can solve it by using ordinary topics.

**Comment by yufei-cai** [comment]
Yes, the optimization works without topic group. Some warning in the documentation would be nice.
